root/etc/e-smith/events/actions/nethserver-dc-user-modify

```diff
@@ -40,3 +46,35 @@ if ($? != 0) {
     die("[ERROR] User $userName modification failed\n");
 }
 
+
+#Create a temporary ldif file (root:root 0600)
+my $fileName = File::Temp->new(
+    UNLINK   => 0,
+    TEMPLATE => "$userName.XXXXXXXXXX",
+    DIR => '/var/lib/machines/nsdc/var/spool/'
+);
+
+#create the ldif content
+my $contentFile = qq (
+dn: cn=$userName,cn=Users,$baseDN
+changetype: modify
+replace: loginShell
+loginShell: $shell
+);
+
+#print the ldif content in the tempfile
+print $fileName $contentFile;
+
+#we need the relative path in the container
+my $relativePathFile = $fileName;
+$relativePathFile =~ s|/var/lib/machines/nsdc||;
+
+#modify the samba LDAP
+system(qq(expect -c 'spawn -noecho /usr/bin/systemd-run -M nsdc -q -t /usr/bin/ldbmodify -H /var/lib/samba/private/sam.ldb $relativePathFile;
+    expect "Modified 1 records successfully" { exit 0 }; exit 3' >/dev/null));
+
+if ($? != 0) {
+    die("[ERROR] User $userName shell modification failed\n");
```

Comment:
Here the temporary file is not cleaned up. I'm not an expert with `File::Temp` module, but from in its documentation page I read about `UNLINK`

> the temporary file is removed by the object destructor if UNLINK is set to true (the default)

I propose to:

- leave the UNLINK parameter to the default
- remove unlink calls
- close the file descriptor to ensure any buffer is flushed

Reply:
It should be good @DavidePrincipi, please comment